library/ssl_tls13_generic.c

```diff
@@ -4643,12 +4643,14 @@ int mbedtls_ssl_parse_new_session_ticket( mbedtls_ssl_context *ssl )
     msg = ssl->in_msg + mbedtls_ssl_hs_hdr_len( ssl );
     msg_len = ( ssl->in_msg[2] << 8 ) | ssl->in_msg[3];
 
-    if( msg_len+ mbedtls_ssl_hs_hdr_len( ssl ) != ssl->in_msglen )
+    if( ssl->in_hslen == 0 || ssl->in_hslen > ssl->in_msglen )
     {
         MBEDTLS_SSL_DEBUG_MSG( 1, ( "bad new session ticket message" ) );
         return( MBEDTLS_ERR_SSL_BAD_HS_NEW_SESSION_TICKET );
     }
 
+    ssl->in_msglen -= ssl->in_hslen;
```

Comment:
This seems like a dangerous change, because we change internal fields of the current messaging layer.

@hannestschofenig What's the rationale for this change?